spec/models/concerns/product_stock_spec.rb

```diff
@@ -0,0 +1,45 @@
+require 'spec_helper'
+
+describe ProductStock do
+  let(:product) { create(:simple_product) }
+
+  context 'when product has no variants' do
+    before do
+      product.variants.first.destroy
+      product.variants.reload
+    end
+
+    scenario 'product.on_demand is master.on_demand' do
```

Comment:
why don't we use `it` like all other unit tests? I'd rather keep the consistency to avoid confusing new devs.